lib/zip/file.rb

```diff
@@ -402,7 +402,7 @@ def get_entry(entry)
     # Creates a directory
     def mkdir(entryName, permissionInt = 0o755)
       raise Errno::EEXIST, "File exists - #{entryName}" if find_entry(entryName)
-      entryName = entryName.dup.to_s
+      entryName = +entryName.to_s
```

Comment:
```suggestion
      entryName = entryName.dup.to_s
```
I am not sure about this one --- I think the `dup` here is not necessarily to get an unfrozen `entryName` but instead to avoid mutating the `entryName` argument. `+` does not duplicate the string if it's not frozen.

Reply:
@jaredbeck ,
Thank you for your comment !
OK, I will revert this change.

Reply:
Done!